scripts/metavar.py

```diff
@@ -1869,8 +1892,21 @@ def declare_variables(self, outfile, indent, dummy=False,
                                      any_scope=False)
             if self.include_var_in_list(var, std_vars=std_vars,
                                         loop_vars=loop_vars, consts=consts):
-                self[standard_name].write_def(outfile, indent, self,
-                                              dummy=dummy)
+                if (not var.get_prop_value('optional')):
+                    self[standard_name].write_def(outfile, indent, self,
+                                                  dummy=dummy)
+                # end if
```

Comment:
Can the code block below (lines 1901-1909) be incorporated into this if block? Or am I missing the reason we have to do a second loop through self.keys()?

```suggestion
                if (not var.get_prop_value('optional')):
                    self[standard_name].write_def(outfile, indent, self,
                                                  dummy=dummy)
                else:
                    self[standard_name].write_def(outfile, indent, self,
                                                  dummy=dummy)
                # end if
```

Reply:
Sorry, we don't need this logical in declare_variables.
It was a relic. Removed.